code/modules/reagents/Chemistry-Reagents/Pyrotechnic-Reagents.dm

```diff
@@ -200,3 +200,47 @@
 		holder.chem_temp += 10
 		holder.handle_reactions()
 	..()
+
+
+/datum/reagent/cryogenic_fluid
+	name = "Cryogenic Fluid"
+	id = "cryogenic_fluid"
+	description = "Extremely cold superfluid used to put out fires that can freeze people solid, chills people when ingested and has a volume dependant freeze reaction on touch."
+	color = "#b3ffff" // rgb: 0, 255, 255
+	metabolization_rate = 2
+
+/datum/reagent/cryogenic_fluid/on_tick()
+	holder.chem_temp -= 5
+	..()
+/datum/reagent/cryogenic_fluid/reaction_mob(mob/living/M, method=TOUCH, reac_volume, show_message = 1)
+	if(iscarbon(M) && M.stat != DEAD)
+		if(method in list(INGEST,INJECT))
+			M.adjust_fire_stacks(-(reac_volume))
+			M.adjustStaminaLoss(reac_volume)
+			M.bodytemperature -= 200
```

Comment:
Those are definitions dude, they can be used for anything.
Things shouldn't be going below absolute zero because it's absolute zero. And just because you can't think of a case that is harmful doesn't mean it is certain to be okay. Even if there is no issue now, there could be an issue down the line.

Also justifying bad code because other code is bad isn't the right attitude either.
Just add a check to make sure the mob isn't going below.
M.bodytemperature = max(M.bodytemperature - 200, TCMB)